src/toxcore/dht/client.rs

```diff
@@ -135,30 +135,38 @@ impl Client {
                 IfAddr::V4(ref addr) => addr.broadcast,
                 _ => None,
         })
-        .map(|ipv4|
-            IpAddr::V4(ipv4)
-        ).collect()
+        .map(|addr|
+            IpAddr::V4(addr)
+        )
+        .collect()
     }
     // get broadcast addresses for Ipv4 and Ipv6
-    fn get_broadcast_addresses() -> Vec<IpAddr> {
-        let mut res = Vec::new();
-        let ipv4s = Client::get_ipv4_broadcast_addrs();
+    fn get_broadcast_addresses(sock_family: u8) -> Vec<IpAddr> {
+        let mut res = Vec::<IpAddr>::new();
 
-        res.extend(ipv4s.iter());
+        // there will be a cases which are host's interface may have IPv4 only,
+        // IPv6 only, both IPv4 and IPv4
+        // If ipv4 only, broadcast to only Ipv4
+        // If ipv6 only, braodcast to only Ipv6
+        // If both, braodcast to both addresses
+        let ips = Client::get_ipv4_broadcast_addrs();
+        res.extend(ips.iter());
 
         // Ipv4 global broadcast address
         res.push(
             ("255.255.255.255").parse().unwrap()
         );
-        // Ipv6 broadcast address
-        res.push(
-            ("FF02::1").parse().unwrap()
-        );
+        if sock_family == 1 {
+            // Ipv6 broadcast address
+            res.push(
+                ("::1").parse().unwrap() // my LAN config has no route to address of FF02::1
+            );
+        }
         res
     }
     /// send LanDiscovery packets to broadcast addresses
-    pub fn send_lan_discovery(&self) -> IoFuture<()> {
-        let ip_addrs = Client::get_broadcast_addresses();
+    pub fn send_lan_discovery(&self, sock_family: u8) -> IoFuture<()> {
```

Comment:
as I told you before, there should be 2 methods: `send_lan_discovery_ipv4` and `send_lan_discovery_ipv6`. Based on the address it is examples/dht_node responsibility to send correct packet. You must not add a new flag and keep it inside Server